aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -113,6 +113,20 @@ boolean keepTokenRefreshed() {
         OAuth2Properties.TOKEN_REFRESH_ENABLED_DEFAULT);
   }
 
+  @Nullable
+  private ScheduledExecutorService tokenRefreshExecutor() {
+    if (!keepTokenRefreshed()) {
+      return null;
+    }
+
+    if (null == TOKEN_REFRESH_EXECUTOR.get()) {
+      TOKEN_REFRESH_EXECUTOR.compareAndSet(
```

Comment:
should this be in synchronous block?

Reply:
this is the main advantage of having an `AtomicReference`: we don't need `synchronized` here because `TOKEN_REFRESH_EXECUTOR` will only be atomically set to the given updated value if the current value is `null`. 
If another thread happened to set `TOKEN_REFRESH_EXECUTOR` first, then we don't do anything here.